### ISSUE TITLE
s3: proper way to mongorestore

### ISSUE THREAD

**Issue by zaklampert** [CollectionFS/Meteor-CollectionFS#697]
Everytime I run a mongorestore, my s3 bucket reverts back to some earlier state - not all files are deleted, but most are. 

What's the proper way to restore a db backup without overwriting all the s3 data? Should I exclude some collection?

**Comment by rhyslbw** [comment]
Sounds like the backup is removing documents from your collection? If so isn't the removal of store data in S3 desired?

**Comment by strikeout** [comment]
I had the same problem, almost hit the production server. I guess its some cleanup logic in cfs that tries to remove files not referenced anymore?

**Comment by jeffyamada** [comment]
Yep, I have this problem too, multiple environments are using the same store, and data being pushed around between them results in vanished files.

@strikeout did you just overwrite this method?

**Comment by jeffyamada** [comment]
Hey guys - just FYI, looks like I solved this by using --excludeCollection=cfs.[your-collection].filerecord so file collections don't get overwritten. One thing to note however is that you'll probably want to organize a couple instances of storage locations so production files aren't so easily manipulated by non-production ones etc. You can then write specific scripts to move the files collections to and from as well. Not the nicest solution, but will be fine once migration scripts are updated accordingly.
